src/lib/parseTableAttributes.ts

```diff
@@ -0,0 +1,56 @@
+/**
+ * DynamoDB Toolbox: A simple set of tools for working with Amazon DynamoDB
+ * @author Jeremy Daly <jeremy@jeremydaly.com>
+ * @license MIT
+ */
+
+import { TableAttributes, TableAttributeConfig, ParsedTableAttribute, DynamoDBTypes } from '../classes/Table'
+
+import { error, typeError, keyTypeError, validTypes, validKeyTypes } from './utils'
+
+// Parse the attributes and verify valid types
+export default (attrs: TableAttributes, partitionKey: string, sortKey: string | null) => 
+  Object.keys(attrs).reduce((acc,field) => {
+
+    if (typeof attrs[field] === 'string') {
+      // TODO: Calling .toString() to satisfy TS even though I'm type checking
+      // Error if invalid key type
+      if ([partitionKey,sortKey].includes(field) && !validKeyTypes.includes(attrs[field].toString())) { keyTypeError(field) }
+      // Error if invalid type
+      if (!validTypes.includes(attrs[field].toString())) { typeError(field) }
+      // Merge and return parsed attribute
+      return Object.assign(acc,parseAttributeConfig(field ,{ type: attrs[field] } as TableAttributeConfig))
+    } else {
+      // TODO: Is there a better way to do this without casting this value?
```

Comment:
If you do an else if and make the comparator instanceof this would be "better" (though it doesn't really matter to the built fist)